packages/headless/src/features/folding/folding-slice.ts

```diff
@@ -172,23 +172,12 @@ export const foldingReducer = createReducer(
       .addCase(
         loadCollection.fulfilled,
         (state, {payload: {collectionId, results}}) => {
-          const rootResult = resolveRootFromFields(
+          const newCollections = createCollections(
             results as ResultWithFolding[],
             state.fields
           );
-          if (!rootResult) {
-            return;
-          }
-          state.collections[collectionId] = {
-            result: rootResult,
-            children: resolveChildrenFromFields(
-              rootResult,
-              results as ResultWithFolding[],
-              state.fields
-            ),
-            moreResultsAvailable: false,
-            isLoadingMoreResults: false,
-          };
+          state.collections[collectionId] = newCollections[collectionId];
+          state.collections[collectionId].moreResultsAvailable = false;
         }
```

Comment:
Is `isLoadingMoreResults` being set to `false` after the refactor?

I did not see any tests in `folding-slice` checking the values of this property and `moreResultsAvailable`.
If there really are no assertions, it would be good to add them to defend against regressions,

Reply:
It's set to false by `createCollections`.

Yeah, ill add more tests around that.